sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -0,0 +1,292 @@
+import numpy as np
+from scipy.sparse import csr_matrix
+from scipy.linalg import block_diag
+from sklearn.decomposition import LatentDirichletAllocation
+
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import raises
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_almost_equal
+from sklearn.utils.testing import assert_greater_equal
+from sklearn.utils.testing import if_not_mac_os
+
+from sklearn.utils.validation import NotFittedError
+from sklearn.externals.six.moves import xrange
+
+
+def _build_sparse_mtx():
+    # Create 3 topics and each have 3 words
+    n_topics = 3
+    doc_topic_prior = 1. / n_topics
+    topic_word_prior = 1. / n_topics
+    block = n_topics * np.ones((3, 3))
+    blocks = [block] * n_topics
+    X = block_diag(*blocks)
+    X = csr_matrix(X)
+    return (n_topics, doc_topic_prior, topic_word_prior, X)
+
+
+def test_lda_fit_batch():
+    """
+    Test LDA batch learning_offset (`fit` method with 'batch' learning)
+    """
+    rng = np.random.RandomState(0)
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior,
+                                    learning_method='batch', random_state=rng)
+    lda.fit(X)
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
```

Comment:
you could additionally check ``adjusted_rand_score([0, 0, 0, 1, 1, 1, 2, 2, 2], np.argmax(lda.transform(X), axis=1) == 1``